dimod/typing.py

```diff
@@ -48,67 +46,127 @@
 
 # use float for python types, https://www.python.org/dev/peps/pep-0484/#the-numeric-tower
 # exclude np.complexfloating from numpy types
-Bias = Union[float, np.floating, np.integer]
+Bias = typing.Union[float, np.floating, np.integer]
+"""A :obj:`~typing.Union` representing objects that can be used as biases.
+
+This includes:
+
+* Python's :class:`int` and :class:`float`.
+* NumPy's :class:`~numpy.floating` and :class:`~numpy.integer`.
+
+"""
+
+# it would be nice to be able to exclude None from Variable. We can define
+# a special abstract base class that excludes None, but making it behave like
+# a typing class is a pain accross our supported Python versions. In the future
+# we should handle it
+
+Variable = typing.Hashable
```

Comment:
I'm wondering whether the uglier `VariableLabel` would help reduce confusion between QMs representing variables and labels of variables in the models...

Reply:
Yes. I had suggested that a while ago as you'll recall. I am open to it, though probably as a followup PR

Reply:
Made an issue https://github.com/dwavesystems/dimod/issues/1160